aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,596 @@
+        &gas_schedule.into_btree_map(),
+        gas_feature_version,
+    )
+    .map_err(|err| {
+        anyhow!(
+            "failed to construct gas params at gas version {}: {}",
+            gas_feature_version,
+            err
+        )
+    })?;
+    let txn_gas_params = &mut gas_params.vm.txn;
+    // The gas schedule needs to be patched in order to allow alternative execution limits
+    // for governance scripts.
+    //
+    // In production we have a mechanism to do this automatically, but that depends on the
+    // scripts' hashes being added to the list of approved execution hashes, which gets skipped
+    // as we skip the governance process altogether.
+    //
+    // TODO: In the future, consider adding the execution hashes of the scripts to the
+    // approval list instead.
+    txn_gas_params.max_execution_gas = txn_gas_params.max_execution_gas_gov;
+    txn_gas_params.max_io_gas = txn_gas_params.max_io_gas_gov;
+    txn_gas_params.max_storage_fee = txn_gas_params.max_storage_fee_gov;
+    state_view.set_state_value(
+        StateKey::resource(&AccountAddress::ONE, &StructTag {
+            address: AccountAddress::ONE,
+            module: Identifier::new(GasScheduleV2::MODULE_IDENTIFIER).unwrap(),
+            name: Identifier::new(GasScheduleV2::TYPE_IDENTIFIER).unwrap(),
+            type_args: vec![],
+        })?,
+        StateValue::new_legacy(
+            bcs::to_bytes(&GasScheduleV2 {
+                feature_version: gas_feature_version,
+                entries: gas_params.to_on_chain_gas_schedule(gas_feature_version),
+            })?
+            .into(),
+        ),
+    );
+
+    // Execute the governance scripts in sorted order.
+    println!("Executing governance scripts...");
+
+    const DUMMY_PROPOSAL_ID: u64 = u64::MAX;
+
+    for (script_idx, (script_path, script_blob)) in scripts.iter().zip(compiled_scripts).enumerate()
+    {
+        // Patch framework functions to skip the governance process.
+        // This is redone every time we execute a script because the previous script could have
+        // overwritten the framework.
+        let features = Features::fetch_config(&state_view)
+            .ok_or_else(|| anyhow!("failed to fetch feature flags"))?;
+        let deserializer_config = aptos_prod_deserializer_config(&features);
+
+        patch_create_signer(&state_view, &deserializer_config)?;
+        // If the script is the last step of the proposal, it MUST NOT have a next execution hash.
+        // Set the boolean flag to true to use a modified patch to catch this.
+        let forbid_next_execution_hash = script_idx == scripts.len() - 1;
+        patch_resolve_multistep_proposal(
+            &state_view,
+            &deserializer_config,
+            forbid_next_execution_hash,
+        )?;
+
+        let script_name = script_path.file_name().unwrap().to_string_lossy();
+        println!("    {}", script_name);
+
+        // Create a new VM to ensure the loader is clean.
```

Comment:
Have you checked this? Because I think warm vm cache does load PackageMetadata for core packages to see if it has changed... or this is the reason, we patch code and package metadata is the same?